### PR TITLE
Home page data

### PR DESCRIPTION
**Pull request by MansijMishra**
Data on the home page is no longer static. 

Updates:

- Username is now dynamically shown depending on who is logged in.
- Activity section renders all food for the given day using the food component.
- Switching dates on the calendar carousel will show the corresponding activity of that day.

To dos: 

- The calories and macros section are still not updating on date switches, this will require an update to the get-daily-metrics api to support searching by date.
- Food items only have dynamic name and calorie data. At the moment, quantity and meal type are not supported. This will have to be addressed in the database fist and then data can be retrieved.

### CODE REVIEW

File: topics-project/app/(main)/home/page.js [hunk 1]
```diff
@@ -1,21 +1,60 @@
...
+  const carouseldate = format(selectedDate, 'MM-dd-yyyy');
+    
   const { calories, fat, carbs, protein, loading, error } = useFetchMetrics(userID);
-  const { proteinGoal, fatGoal, carbsGoal } = useFetchUserInfo(userID);
+  const { proteinGoal, fatGoal, carbsGoal, userName } = useFetchUserInfo(userID);
+
+  const NutritionList = () => {
+  const { nutritionEntries, loading, error } = useFetchNutritionByDate(userID, carouseldate);
+
+  if (loading) return <p>Loading...</p>;
```
Comment (Abdul-Andha) [inline topics-project/app/(main)/home/page.js:41]: Do we want to use the spinner component here instead?

**Review by Abdul-Andha — COMMENTED** [review]
Looks good, I just had one question. Also, let's add tickets in the backlog for the new TODOs you mentioned!

**Review by olukukoyi — COMMENTED** [review]
lgtm

File: topics-project/app/(main)/home/page.js (hunk 1 above)
Comment (MansijMishra) [reply topics-project/app/(main)/home/page.js:41]: Will be addressing this in the next PR, thanks for the suggestion!

**Pull request merged**